### ISSUE TITLE
Override Manifest cache policy

### ISSUE THREAD

**Issue by Ernest0-Production** [tuist/tuist#6245]
### What problem or need do you have?

I want to be able to control the input data for caching the project manifest.

**Motivation:**

In the manifest files I have logic that requires reading the current directory at the time of generation.
This is required to automate some actions in my project that are not available in tuist at the moment, and it hardly makes sense to include them in tuist itself, because There can be many such scenarios in different projects.

### Potential solution

In the simplest case, it makes sense to add the ability to completely disable the manifest cache, without using the command `tuist clean manifests` which clears all manifests of all projects on my computer.

Ideally, the ability to specify a list of additional files for which checksum will be calculated.

For example, add a parameter to `Config`:
```swift
let config = Config(
   manifestCache: .enable( 
       extending: [Path], // Support glob path
  )
)

let config = Config(
   manifestCache: .disabled
)
```

### macOS version

15

### Tuist version

4.10

### Xcode version

15.3

**Comment by pepicrft** [comment]
@Ernest0-Production 
Manifests that contain logic that depends on the current directory is generally discouraged

> This is required to automate some actions in my project

Could you share an example so that I can provide with a solution or path forward.

**Comment by Ernest0-Production** [comment]
> Could you share an example so that I can provide with a solution or path forward

Hello. I really wanted to be able to write a custom mechanism for automatically generating type-safe access to target resources via SwiftGen.

The limited API of Tuist with the `resource synthesizer` did not really suit me, and I decided to add my own side effects to the project generation (`tuist generate`):
- in the `Project.swift` file, when creating a target and if resources were available, I generated a `ResourceType+Generated.swift` file.
```swift
extension Target { 
}
```
- Before generating the project, I generated a checksum of the project file structure. If ResourceType+Generated.swift was deleted, and/or a new resource type appeared (localization or fonts, for example), then the user would invalidate the cache and recompile and run the manifest, causing my side effects to be called again.

I came to this decision because I tried many other right and wrong approaches. The reason for this is that it is important for me to have access to the code that I use to generate the project.
The ProjectAutomation API is very limited and does not allow you to change the project. And using XcodeProj forces you to go down to a lower level, which does not seem very convenient.
